neurobooth_os/gui.py

```diff
@@ -64,6 +64,54 @@ def _process_received_data(serv_data, window):
             window.write_event_value(event, f"{stream_name}, {filename}]")
             
 
+def _find_subject(window, conn, first_name, last_name):
+    """Find subject from database"""
+    subject_df = meta.get_subject_ids(conn, first_name, last_name)
+    window['dob'].update(values=subject_df['date_of_birth'])
+    return subject_df
+
+
+def _select_subject(window, subject_df):
+    """Select subject from the DOB window"""
+    subject = subject_df.iloc[window['dob'].get_indexes()]
+    subject_id = subject.name
+    first_name = subject['first_name_birth']
+    last_name = subject['last_name_birth']
+
+    # Update GUI
+    window['dob'].update(values=[f'Subject ID: {subject_id}'])
+    window['first_name'].update(disabled=True)
+    window['last_name'].update(disabled=True)
+    return first_name, last_name, subject_id
+
+
+def _get_tasks(window, conn, collection_id):
+    tasks_obs = meta.get_tasks(collection_id, conn)
+    tasks = list()
+    for task in tasks_obs:
+        task_id, *_ = meta._get_task_param(task, conn)
+        tasks.append(task_id)
+    window["_tasks_"].update(value=", ".join(tasks))
+    return tasks
+
+
+def _get_collections(window, conn, study_id):
+    collection_ids = meta.get_collection_ids(study_id, conn)
+    window["collection_id"].update(values=collection_ids)
+    return collection_ids
+
+
+def _present_tasks(window, tasks, subject_id, steps, nodes):
```

Comment:
`nodes` should really be a dictionary.

```py
    if remote:
        database = "mock_neurobooth"
        nodes = dict(acquisition='dummy_acq', presentation='dummy_stm')
        host_ctr, port_ctr = node_info("dummy_ctr")
    else:
        nodes = dict(acquisition='acquisition', presentation='presentation')
        host_ctr, port_ctr = node_info("control")
```

then you can say `nodes['presentation']` instead of `nodes[1]` making it more readable